File: content/atlas/source/includes/avs/index-management/update-index/EditAutoEmbedIndex.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
import org.bson.conversions.Bson;

import java.util.Arrays;

public class EditVectorIndex {

    public static void main(String[] args) {

        // Replace the placeholder with your connection string
        String uri = "<connectionString>";

        // Connect to your cluster
        try (MongoClient mongoClient = MongoClients.create(uri)) {

            // Set the namespace
            MongoDatabase database = mongoClient.getDatabase("<databaseName>");
            MongoCollection<Document> collection = database.getCollection("<collectionName>");

            // Define the index changes
            String indexName = "<indexName>";
            Bson definition = new Document(
                "fields",
                Arrays.asList(
                    new Document("type", "autoEmbed")
                        .append("modality", "text")
                        .append("model", "<modelName>")
                        .append("path", "<fieldToIndex>"),
                    new Document("type", "filter")
                        .append("path", "<filterField1>"),
                    new Document("type", "filter")
                        .append("path", "<filterField2>")));

            // Update the index
            collection.updateSearchIndex(indexName, definition);
            System.out.println("Successfully updated the index");
        }
    }
}

File: content/atlas/source/includes/avs/index-management/update-index/edit-auto-embed-index.go
package main  
  
import (  
	"context"  
	"fmt"  
	"log"  
  
	"go.mongodb.org/mongo-driver/v2/mongo"  
	"go.mongodb.org/mongo-driver/v2/mongo/options"  
)  
  
func main() {  
	ctx := context.Background()  
  
	// Replace the placeholder with your connection string  
	const uri = "<connection-string>"  
  
	// Connect to your cluster  
	clientOptions := options.Client().ApplyURI(uri)  
	client, err := mongo.Connect(ctx, clientOptions)
	if err != nil {  
		log.Fatalf("failed to connect to the server: %v", err)  
	}  
	defer func() { _ = client.Disconnect(ctx) }()  
  
	// Set the namespace  
	coll := client.Database("<database-name>").Collection("<collection-name>")  
	indexName := "<index-name>"  
  
	type autoEmbedField struct {  
		Type     string `bson:"type"`  
		Modality string `bson:"modality"`  
		Path     string `bson:"path"`  
		Model    string `bson:"model"`  
	}  
  
	type filterField struct {  
		Type string `bson:"type"`  
		Path string `bson:"path"`  
	}  
  
	type indexDefinition struct {  
		Fields []interface{} `bson:"fields"`  
	}  
  
	definition := indexDefinition{  
		Fields: []interface{}{  
			autoEmbedField{  
				Type:     "autoEmbed",  
				Modality: "text",  
				Path:     "<indexed-field>",  
				Model:    "<embedding-model>",  
			},  
			filterField{  
				Type: "filter",
				Path: "<field-to-index>",  
			},  
		},  
	}  
  
	err = coll.SearchIndexes().UpdateOne(ctx, <index-name>, definition)  
  
	if err != nil {  
		log.Fatalf("failed to update the index: %v", err)  
	}  
  
	fmt.Println("Successfully updated the search index")  
}  